flutter_frontend_server/lib/server.dart

```diff
@@ -169,3 +182,92 @@ Future<int> starter(
   frontend.listenAndCompile(compiler, input ?? stdin, options, completer);
   return completer.future;
 }
+
+/// A [RecursiveVisitor] that replaces [Object.toString] overrides with
+/// `super.toString()`.
+class ToStringVisitor extends RecursiveVisitor<void> {
+  /// The [packageUris] must not be null.
+  ToStringVisitor(this._packageUris) : assert(_packageUris != null);
+
+  /// A set of package URIs to apply this transformer to, e.g. 'dart:ui' and
+  /// 'package:flutter/foundation.dart'.
+  final Set<String> _packageUris;
+
+  /// Turn 'dart:ui' into 'dart:ui', or
+  /// 'package:flutter/src/semantics_event.dart' into 'package:flutter'.
+  String _importUriToPackage(Uri importUri) => '${importUri.scheme}:${importUri.pathSegments.first}';
+
+  bool _isInTargetPackage(Procedure node) {
+    return _packageUris.contains(_importUriToPackage(node.enclosingLibrary.importUri));
+  }
+
+  bool _hasKeepAnnotation(Procedure node) {
+    for (ConstantExpression expression in node.annotations.whereType<ConstantExpression>()) {
+      if (expression.constant is! InstanceConstant) {
+        continue;
+      }
+      final InstanceConstant constant = expression.constant as InstanceConstant;
+      if (constant.classNode.name == 'pragma' && constant.classNode.enclosingLibrary.importUri.toString() == 'dart:core') {
+        final Reference key = constant.fieldValues.keys.firstWhere((Reference ref) {
+          return ref.asField?.name?.name == 'name';
+        });
+
+        final Constant pragmaName = constant.fieldValues[key];
+        if (pragmaName is! StringConstant) {
+          continue;
+        }
+        if ((pragmaName as StringConstant).value == 'flutter_frontend:keep') {
```

Comment:
@jonahwilliams @zanderso wdyt?

Reply:
Could you give an example of where this annotation would go? In what situations is it needed?

Reply:
```dart
class MyStringBuffer {
  StringBuffer _stringBuffer = StringBuffer();

  // ...

  @override
  @pragma('flutter_frontend:keep')
  String toString() {
    return _stringBuffer.toString();
  }
}
```

Would prevent the transformer from overwriting the method.

Reply:
Are there cases where a program would be incorrect, or the transformation would do something wrong if the annotation is not present?

Reply:
If someone tried to apply this transfomer to `dart:core`, it would break their program - all the primitive type classes override toString in meaningful ways, as does `StringBuffer`. 

You shouldn't do that, and we shouldn't make it easy for people to do that.  The plan right now is to use this for `dart:ui` and `package:flutter` to save binary space, and to test this in a controlled internal environment with programs that have good test coverage in the modes we apply it to.

Reply:
Talked offline.  Going to add a type to `dart:ui` called `keepToString`.

Reply:
Added annotation to dart:ui for this called `keepToString`, plus documentation there about this feature and how to use the annotation.